src/audiovideocontroller/DefaultAudioVideoController.ts

```diff
@@ -862,6 +868,18 @@ export default class DefaultAudioVideoController
     );
   }
 
+  private updateNumberOfPublishedVideoSources(): void {
+    const selfAttendeeId = this.configuration.credentials.attendeeId;
+    const hasLocalVideo = this._videoTileController.hasStartedLocalVideoTile();
+    if (this.meetingSessionContext.videoStreamIndex !== null) {
+      const videoCount =
+        this.meetingSessionContext.videoStreamIndex.numberOfVideoPublishingParticipantsExcludingSelf(
+          selfAttendeeId
+        ) + (hasLocalVideo ? 1 : 0);
+      this.connectionHealthData.setNumberOfPublishedVideoSources(videoCount);
```

Comment:
For this change it's fine, but theoretically i feel like for future flexibility, 'health data' would separate out send and receive stream count. What do you think?

Reply:
Yeah, the count of receive stream might be a more precise one to determine codec fallback in conditions like this. It does require the mechanism to feedback info from receiver to sender. It will bring back the problem of controlling encoding from decoding and it have to be done in a systematic way (together with feedback of other information). I agree it would be a good improvement for future, but for this stage, I think we can proceed with send stream count.